### PR TITLE
Fix iOS redirection when scheme is https

### LINKED ISSUE #65: Unable to Capture Redirect URLs with 'http' or 'https' CallbackUrlScheme on iOS without user interaction
## Describe the bug
When using this library with Universal Links on iOS to login into a service, the link is opened in the browser if the user is already logged in.

## To Reproduce
1. Setup Universal Links in example app and custom domain (and verify they are valid from iOS Settings -> Developer -> Universal Link)
2. Edit the example app connection html page to include this snippet: 
```html
    <p>Redirecting in 3 seconds...</p>
    <script>
        setTimeout(function() {
            window.location.href = 'https://your.domain.com/?token=balalalala';
        }, 3000); // 3000 milliseconds = 3 seconds
    </script>
```
4. Run the example app, try to login and wait for 3 seconds
5. The universal link is opened inside the browser instead of returning to the app

## Expected behavior
When using a redirect to a universal link without user interaction, browser closes and app receives the callback

## Device (please complete the following information!)
 - Device: iPhone 11
 - OS: iOS 16.6
 - Browser: Safari
 - `flutter_web_auth_2` version: 2.2.1

## Additional context
This seems more like a bug of Safari, instead of this library's code (Read "Preparing Your App to Handle Universal Links" at https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html)
Nevertheless I have lost 2 days trying to fix it so it seemed logic to me that this situation must be shared for future developers. 

## Checklist
 - [x] I have read and followed the **entire** [troubleshooting guide](https://github.com/ThexXTURBOXx/flutter_web_auth_2#troubleshooting) and it has not provided the solution I need.
 - [x] I have provided all the information I can (incl. auth URL etc.)

### LINKED ISSUE #104: Add ASWebAuthenticationSession.Callback support
## Is your feature request related to a problem? Please describe.
If I wanna use the **https scheme** and [Universal Links](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html) on iOS the completionHandler is not called.

- This is related to Issue #65 

## Describe the solution you'd like
Apple has [deprecated the previous initializer](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession/2990952-init) matching in favor of the [new initializer](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession/4316247-init) factory functions:

- `ASWebAuthenticationSession.Callback.customScheme(String)`
- `ASWebAuthenticationSession.Callback.https(host: String, path: String)`

### PR DESCRIPTION
**Pull request by GabrielDamasceno**
Enable `ASWebAuthenticationSession.Callback.https` support to fix Universal Links on iOS.

Fixes #65
Fixes #104

### CODE REVIEW

**Comment by ThexXTURBOXx** [comment]
Thank you very much for your PR, it's very appreciated!
I have reverted some of the styling changes.
Actually, I had individual options (like you had proposed) in my mind in the past, but this will lead to very messy code - both in `flutter_web_auth_2` and on the user's end.
This is why I ultimately did not go that route.
I do have a few more tiny things, though. One second.

File: flutter_web_auth_2/ios/Classes/SwiftFlutterWebAuth2Plugin.swift [hunk 1]
```diff
@@ -64,7 +66,11 @@ public class SwiftFlutterWebAuth2Plugin: NSObject, FlutterPlugin {
             if #available(iOS 12, *) {
                 var _session: ASWebAuthenticationSession? = nil
                 if #available(iOS 17.4, *) {
-                    _session = ASWebAuthenticationSession(url: url, callback: ASWebAuthenticationSession.Callback.customScheme(callbackURLScheme), completionHandler: completionHandler!)
+                    if (callbackURLScheme == "https") {
+                        _session = ASWebAuthenticationSession(url: url, callback: ASWebAuthenticationSession.Callback.https(host: host, path: path), completionHandler: completionHandler!)
```
Comment (ThexXTURBOXx) [inline flutter_web_auth_2/ios/Classes/SwiftFlutterWebAuth2Plugin.swift:70]: Could you please add an `if` check here which throws a "good"/"readable" error message when `callbackURLScheme == "https"`, but `host` is `null` or `path` is `null`?

File: flutter_web_auth_2/macos/Classes/FlutterWebAuth2Plugin.swift [hunk 2]
```diff
@@ -42,7 +44,11 @@ public class FlutterWebAuth2Plugin: NSObject, FlutterPlugin, ASWebAuthentication
 
             var _session: ASWebAuthenticationSession? = nil
             if #available(macOS 14.4, *) {
-                _session = ASWebAuthenticationSession(url: url, callback: ASWebAuthenticationSession.Callback.customScheme(callbackURLScheme), completionHandler: completionHandler)
+                if (callbackURLScheme == "https") {
+                    _session = ASWebAuthenticationSession(url: url, callback: ASWebAuthenticationSession.Callback.https(host: host, path: path), completionHandler: completionHandler)
```
Comment (ThexXTURBOXx) [inline flutter_web_auth_2/macos/Classes/FlutterWebAuth2Plugin.swift:48]: Please add the same (or a similar, if needed) `if` check as above here.

**Review by ThexXTURBOXx — CHANGES_REQUESTED** [review]
And finally, after all of that, could you please check whether everything still compiles correctly and/or works correctly?
Thank you very much!

File: flutter_web_auth_2/ios/Classes/SwiftFlutterWebAuth2Plugin.swift (hunk 1 above)
Comment (GabrielDamasceno) [reply flutter_web_auth_2/ios/Classes/SwiftFlutterWebAuth2Plugin.swift]: Hi, thanks for the quick feedback! I made the adjustments you requested and it seems to work as expected.

**Comment by ThexXTURBOXx** [comment]
The pipeline has passed, so everything seems to be in order now :)
Thank you very much for your contribution - will merge now and publish a new pre-release soon!

**Comment by GabrielDamasceno** [comment]
Thank you for maintaining this plugin ;)

Best regards!

**Pull request merged**